internal/resources/cluster/resource_cluster.go

```diff
@@ -100,6 +102,29 @@ func constructFullname(d *schema.ResourceData) (fullname *clustermodel.VmwareTan
 	return fullname
 }
 
+func constructNpFullName(d *schema.ResourceData) (fullname *nodepoolmodel.VmwareTanzuManageV1alpha1ClusterNodepoolFullName) {
+	fullname = &nodepoolmodel.VmwareTanzuManageV1alpha1ClusterNodepoolFullName{}
+
+	if value, ok := d.GetOk(ManagementClusterNameKey); ok {
+		fullname.ManagementClusterName = value.(string)
+	}
+
+	if value, ok := d.GetOk(ProvisionerNameKey); ok {
+		fullname.ProvisionerName = value.(string)
+	}
+
+	if value, ok := d.GetOk(NameKey); ok {
+		fullname.ClusterName = value.(string)
+	}
+
+	if value, ok := d.Get(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, topologyKey, nodePoolKey)).([]interface{})[0].(map[string]interface{}); ok {
```

Comment:
This looks very specific to TKGs and not supporting TKm vsphere?

Reply:
good catch, i missed that reference.